modules/apps/object/object-service/src/main/java/com/liferay/object/internal/action/trigger/messaging/ObjectActionTriggerMessageListener.java

```diff
@@ -6,55 +6,201 @@
+import java.util.List;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
 
 /**
- * @author Marco Leo
- * @author Brian Wing Shun Chan
+ * @author Marco Leov
  */
+@Component(
+	property = {
+		"destination.name=" + DestinationNames.OBJECT_ENTRY_ATTACHMENT_DOWNLOAD,
+		"destination.name=" + DestinationNames.OBJECT_ENTRY_ON_AFTER_ADD,
+		"destination.name=" + DestinationNames.OBJECT_ENTRY_ON_AFTER_UPDATE
+	},
+	service = MessageListener.class
+)
 public class ObjectActionTriggerMessageListener extends BaseMessageListener {
 
-	public ObjectActionTriggerMessageListener(
-		String className, ObjectActionEngine objectActionEngine,
-		String objectActionTriggerKey) {
+	@Activate
+	protected void activate(BundleContext bundleContext) {
+		_registerService(
+			bundleContext, DestinationNames.OBJECT_ENTRY_ON_AFTER_ADD);
+
+		_registerService(
+			bundleContext, DestinationNames.OBJECT_ENTRY_ON_AFTER_UPDATE);
+
+		_registerService(
+			bundleContext, DestinationNames.OBJECT_ENTRY_ATTACHMENT_DOWNLOAD);
+	}
+
+	@Deactivate
+	protected void deactivate() {
+		_serviceRegistration.forEach(ServiceRegistration::unregister);
 
-		_className = className;
-		_objectActionEngine = objectActionEngine;
-		_objectActionTriggerKey = objectActionTriggerKey;
+		_serviceRegistration.clear();
 	}
 
 	@Override
-	protected void doReceive(Message message) {
+	protected void doReceive(Message message) throws Exception {
+		String objectDefinitionExternalReferenceCode = message.getString(
+			"objectDefinitionExternalReferenceCode");
+		long companyId = message.getLong("companyId");
+
+		ObjectDefinition objectDefinition =
+			_objectDefinitionLocalService.
+				fetchObjectDefinitionByExternalReferenceCode(
+					objectDefinitionExternalReferenceCode, companyId);
+
+		if (objectDefinition == null) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(
+					StringBundler.concat(
+						"Object definition is null for external reference ",
+						"code ", objectDefinitionExternalReferenceCode,
+						" and company ", companyId));
+
+				return;
+			}
+		}
+
+		String objectEntryExternalReferenceCode = message.getString(
+			"objectEntryExternalReferenceCode");
+
+		ObjectEntry objectEntry = _objectEntryLocalService.fetchObjectEntry(
+			objectEntryExternalReferenceCode,
+			objectDefinition.getObjectDefinitionId());
+
+		if (objectEntry == null) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(
+					StringBundler.concat(
+						"Object entry is null for external reference code ",
+						objectEntryExternalReferenceCode,
+						" and object definition ",
+						objectDefinitionExternalReferenceCode));
+
+				return;
+			}
+		}
+
+		String objectActionTrigger = message.getString("actionTrigger");
```

Comment:
The key is inconsistent. Can you please change it to "objectActionTrigger" in all places?